apps/src/dance/ai/AiBlockPreview.tsx

```diff
@@ -0,0 +1,42 @@
+import {BlockSvg, Workspace, WorkspaceSvg} from 'blockly';
+import React, {useEffect, useRef} from 'react';
+import moduleStyles from './ai-block-preview.module.scss';
+
+interface AiPreviewProps {
+  generateBlocksFromResult: (workspace: Workspace) => [BlockSvg, BlockSvg];
+}
+
+/**
+ * Previews the blocks generated by the AI block in Dance Party.
+ */
+const AiBlockPreview: React.FunctionComponent<AiPreviewProps> = ({
+  generateBlocksFromResult,
+}) => {
+  const blockPreviewContainerRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (blockPreviewContainerRef.current) {
+      const emptyBlockXml = Blockly.utils.xml.textToDom('<xml></xml>');
```

Comment:
Is there a way to do this using the newer JSON format, which I think we've now transitioned to?

Reply:
Probably? I do feel like using our wrapper's `createReadOnlyBlockSpace` method seems appropriate here, and it only takes xml. Refactoring it to allow JSON might be a good idea though?

Reply:
No worries if it's out of scope for now, was just curious.